doc/reference/trd104-syscalls.md

```diff
@@ -545,28 +545,26 @@ support multiple allowed buffers.
 The Tock kernel MUST check that the passed buffer is contained within
 the calling process's writeable address space. Every byte of the
 passed buffer must be readable and writeable by the
-process. Zero-length buffers may therefore have abitrary addresses. If
+process. Zero-length buffers may therefore have arbitrary addresses. If
 the passed buffer is not complete within the calling process's
 writeable address space, the kernel MUST return a failure result with
 an error code of `INVALID`.
 
-Because a process relinquishes access to a buffer when it makes a
-Read-Write Allow call with it, the buffer passed on the subsequent
-Read-Write Allow call cannot overlap with the first passed buffer.
-This is because the application does not have access to that
-memory. If an application needs to extend a buffer, it must first call
+When a process passes a buffer to
```

Comment:
Why remove the explanatory clause? I'm open to removing it, but I had included it because I thought it was useful to explain how overlapping would implicitly violate the lack of access (it means userspace had access before).

Reply:
I removed the explanatory clause because I removed the word "access" in favor of "modify" in other places, so it no longer made sense. I am happy to restore it if others agree that "access" is actually what we want here

Reply:
See my longer comment above. I think that if we want userspace to be able to access allowed buffers without calling allow again, we will need to require some coupling between allow buffer numbers and callbacks/commands. I.e., allow buffer X can be accessed in the period between callback Y being issued and command Z being called.